.github/workflows/deploy-instance.yml

```diff
@@ -0,0 +1,57 @@
+name: Deploy instance
+
+on:
+  pull_request:
+    types: [opened, reopened, labeled, synchronize]
+
+# Cancel previous runs of the same workflow and PR number or branch/tag
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  deploy-instance:
+    if: contains(github.event.pull_request.labels.*.name, 'deploy')
+    runs-on: ubuntu-latest
+    env:
+      INSTANCE_HOST: 'https://dev.im.dhis2.org'
+      INSTANCE_NAME: pr-${{ github.event.pull_request.number }}
+    steps:
+      - name: Wait for API tests
+        uses: lewagon/wait-on-check-action@v1.3.1
+        with:
+          ref: ${{ github.head_ref }}
+          check-name: 'api-test'
+          repo-token: ${{ secrets.GITHUB_TOKEN }}
+
+      - uses: actions/checkout@v4
+        with:
+          repository: dhis2-sre/im-manager
+          sparse-checkout: scripts/instances
+
+      - name: Install HTTPie
+        run: python -m pip install httpie
+
+      - name: Deploy DHIS2 instance
+        working-directory: scripts/instances
+        env:
+          HTTP: https --check-status
+          USER_EMAIL: ${{ secrets.IM_BOT_EMAIL }}
+          PASSWORD: ${{ secrets.IM_BOT_PASSWORD }}
+          IM_HOST: 'https://api.im.dhis2.org'
+          IMAGE_REPOSITORY: core-pr
+          IMAGE_TAG: ${{ github.event.pull_request.number }}
+          IMAGE_PULL_POLICY: Always
+          DATABASE_ID: sl-sierra-leone-dev-sql-gz
+        run: ./findByName.sh dev $INSTANCE_NAME && ./restart.sh dev $INSTANCE_NAME || ./deploy-dhis2.sh dev $INSTANCE_NAME
+
+      - name: Wait for instance
+        run: timeout 300 bash -c 'while [[ "$(curl -fsSL -o /dev/null -w %{http_code} $INSTANCE_HOST/$INSTANCE_NAME)" != "200" ]]; do sleep 5; done'
+
+      - name: Generate analytics
+        run: curl -X POST -u 'admin:district' "$INSTANCE_HOST/$INSTANCE_NAME/api/resourceTables/analytics"
```

Comment:
It just occurred to me that we should probably use secrets for the SL login (we might want to use something other than admin:district).

Also, I presume this will fail on fork PRs?

Reply:
Good point about the secrets! And yes, with secrets this will fail on fork PRs, which is okay I suppose?